README.md

```diff
@@ -1,186 +1,149 @@
-
-![Cocostuff results](results/figures/cocostuff27_results.jpg)
-
-
-## Citation
+For PCA and RP (and HNNE in the appendix), we implemented a mechanism that caches the DINO features using a [Squirrel](https://github.com/merantix-momentum/squirrel-core) data loader. This approach saves lots of compute because the DINO features stay the same regardless of the target dimension we project the features into. Having pre-processed DINO features also speeds the ablations up by a large margin. The pre-processed datasets may become very large, so you might want to save your data to a remote data storage (e.g., GCS bucket or AWS S3) by setting `root_feat_store` to a remote path that [fsspec](https://filesystem-spec.readthedocs.io/en/latest/) can handle. Now, execute the below script, which will cache the DINO features of all datasets under `root_feat_store`.
 
+```bash
+python mxm_precompute_dino_feats.py batch_size=64 # better system util with higher batch size
 ```
-@inproceedings{hamilton2022unsupervised,
-	title={Unsupervised Semantic Segmentation by Distilling Feature Correspondences},
-	author={Mark Hamilton and Zhoutong Zhang and Bharath Hariharan and Noah Snavely and William T. Freeman},
-	booktitle={International Conference on Learning Representations},
-	year={2022},
-	url={https://openreview.net/forum?id=SaKO6z6Hl0c}
-}
+
+We wrote a variant `mxm_train_segmentation.py` of the original `train_segmentation.py` script that uses the pre-computed DINO features instead of running the usual DINO forward pass. It was easier to copy the original `train_segmentation.py` and create a new `mxm_train_segmentation.py` with these more invasive additions instead of editing the original file and keeping it backward compatible. 
+
+Now you can run the remaining benchmarks on the other dimension reduction baselines. Once you start the script, the dimension reduction techniques (i.e., PCA, RP) will be automatically fitted on a subset of the training dataset. For HNNE, we need not only fit the method but also pre-compute its down-projected features because calculating them on the fly in the model's forward pass is prohibitively expensive. Hence, to reproduce the HNNE results from the appendix, run `python mxm_dimred.py dataset_name="cityscapes"`. Consequently, the training script will directly fetch the pre-computed, lower-dimensional HNNE features.
+
+```bash
+# COCOSTUFF
+for dimred in "PCA" "RP":
```

Comment:
Quick tip for next time: in bash, everything is a string, so there is no need to include the `"` (unless there are some escape characters involved...). Also, you can then access the variable in a  string context as 
```bash
${dimred}_coco_dim${dim}_date${DATE}
```
so you don't need the `"` here either which is more bashonic 🙃